src/components/delete-trace.tsx

```diff
@@ -0,0 +1,48 @@
+import { TrashIcon } from "lucide-react";
+import { useRouter } from "next/router";
+
+import { Button } from "@/src/components/ui/button";
+import { api } from "@/src/utils/api";
+
+export function DeleteTrace({
+  traceId,
+  projectId,
+  icon,
+}: {
+  traceId: string,
+  projectId: string,
+  icon?: boolean,
+}
+) {
+  const router = useRouter();
+
+  const mutDeleteTrace = api.traces.delete.useMutation();
+
+  const deleteTrace = async () => {
+    await mutDeleteTrace.mutateAsync({traceId: traceId});
+    if (icon) {
+      router.reload();
```

Comment:
Do not reload but use invalidation of trpc cache, thereby the table updates without a full refresh.

Reply:
ok... no problem